main.go

```diff
@@ -1,43 +1,75 @@
 package main
 
 import (
+	"errors"
 	"log"
 	"net/http"
+	"net/http/httputil"
+	"net/url"
 	"observability-remote-write-proxy/pkg/remotewrite"
+	"os"
+)
+
+const (
+	prefixHeader = "X-Forwarded-Prefix"
+)
+
+var (
+	PORT     = os.Getenv("PORT")
+	upstream = os.Getenv("UPSTREAM")
 )
 
 func main() {
-	handler := http.HandlerFunc(handleRequestSuccess)
-	http.Handle("/", handler)
-	err := http.ListenAndServe(":8080", nil)
+	upstreamUrl, err := url.Parse(upstream)
+	if err != nil {
+		panic(err)
+	}
+
+	proxy := httputil.ReverseProxy{
+		Director: func(request *http.Request) {
+			request.URL.Scheme = upstreamUrl.Scheme
+			request.Host = upstreamUrl.Host
+			request.URL.Host = upstreamUrl.Host
+			request.URL.Path = upstreamUrl.Path
+			request.Header.Add(prefixHeader, "/")
+		},
+	}
+
+	proxy.Transport = http.DefaultTransport
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
at some point we could have a:

`/ready` and `live` handlers for readness and liveness checks.

Reply:
/cc @valeriiashapoval @pb82 

What do you think?

Reply:
Yes, we can add that. Or even a metrics endpoint for proxy metrics.

Reply:
sounds good @machi1990